src/atom-polymer.ts

```diff
@@ -33,8 +33,9 @@ class AtomPolymer {
 
   deactivate() {
     this.subscriptions.dispose();
-    this.linter = null;
-    this.autocompleter = null;
+    this.editorService = null;
+    this.linter = new Linter();
+    this.autocompleter = new Autocompleter();
```

Comment:
+1, good point. This is easier to understand and easier to reason about. I was worried that provideLinter or provideAutocompleter might be called before activation, but in hindsight that seems like it totally shouldn't happen.